ludwig/schema/features/augmentation/utils.py

```diff
@@ -0,0 +1,145 @@
+from dataclasses import field
+
+from marshmallow import fields, ValidationError
+
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.constants import TYPE
+from ludwig.schema import utils as schema_utils
+from ludwig.utils.registry import Registry
+
+_augmentation_config_registry = Registry()
+
+
+@DeveloperAPI
+def get_augmentation_config_registry() -> Registry:
+    return _augmentation_config_registry
+
+
+def register_augmentation_config(name: str):
+    def wrap(cls):
+        get_augmentation_config_registry()[name] = cls
+        return cls
+
+    return wrap
+
+
+def get_augmentation_cls(name: str):
+    return get_augmentation_config_registry()[name]
+
+
+@DeveloperAPI
+def AugmentationContainerDataclassField(default=[], description=""):
+    """Custom dataclass field that when used inside a dataclass will allow the user to specify an augmentation
+    config.
+
+    Args:
+        default: The default augmentation config to use.
+        description: The description of the augmentation config.
+
+    Returns: Initialized dataclass field that converts an untyped dict with params to an augmentation config.
+    """
+
+    class AugmentationContainerMarshmallowField(fields.Field):
+        """Custom marshmallow field that deserializes a list for a valid augmentation config from the
+        augmentation_registry and creates a corresponding JSON schema for external usage."""
+
+        def _deserialize(self, value, attr, data, **kwargs):
+            assert isinstance(value, list), "Augmentation config must be a list."
+
+            augmentation_list = []
+            for augmentation in value:
+                augmentation_op = augmentation[TYPE]
+                augmentation_cls = get_augmentation_cls(augmentation_op)
+                pre = augmentation_cls()
+                try:
+                    augmentation_list.append(pre.Schema().load(augmentation))
+                except (TypeError, ValidationError) as error:
+                    raise ValidationError(
+                        f"Invalid augmentation params: {value}, see `{pre}` definition. Error: {error}"
+                    )
+            return augmentation_list
+
+        @staticmethod
+        def _jsonschema_type_mapping():
+            return get_augmentation_jsonschema()
+
+    try:
+        if default:
```

Comment:
I would add the assert `isinstance(default, list)` here as well

Reply:
done